stable_baselines3/common/monitor.py

```diff
@@ -38,33 +41,26 @@ def __init__(
     ):
         super(Monitor, self).__init__(env=env)
         self.t_start = time.time()
-        if filename is None:
-            self.file_handler = None
-            self.logger = None
+        if filename:
+            self.results_writer = ResultsWriter(
+                filename,
+                header={"t_start": self.t_start, "env_id": env.spec and env.spec.id},
```

Comment:
why `env.spec and env.spec.id` ?

Reply:
This is largely taken from `openai/baselines`. I am not very familiar with the CSV related workflow.

Reply:
now I get it, only try to access `env.spec.id`, if `env.spec` is not None, otherwise it would throw an error